pkg/actuators/machine/actuator.go

```diff
@@ -347,17 +347,20 @@ func (a *Actuator) DeleteMachine(machine *machinev1.Machine) error {
 		return a.handleMachineError(machine, err, deleteEventAction)
 	}
 
-	instances, err := getRunningInstances(machine, client)
+	// Get all instances not terminated.
+	existingInstances, err := a.getMachineInstances(machine)
 	if err != nil {
-		glog.Errorf("%s: error getting running instances: %v", machine.Name, err)
+		glog.Errorf("%s: error getting existing instances: %v", machine.Name, err)
```

Comment:
How do we resolve the conflict this creates with #290? Within #290 once this is merged is probably easiest right? cc @alexander-demichev

Reply:
thanks! let's hold the other ones and let this move forwards as it's an important bug fix that we want to back port to 4.3 right away.